src/utils.rs

```diff
@@ -75,3 +109,93 @@ pub fn resolve_url<T: AsRef<str>, U: AsRef<str>>(from: T, to: U) -> Result<Strin
     };
     Ok(result)
 }
+
+pub fn resolve_css_imports(
+    cache: &mut HashMap<String, String>,
+    css_string: &str,
+    as_dataurl: bool,
+    href: &str,
+    opt_no_images: bool,
+    opt_user_agent: &str,
+    opt_silent: bool,
+    opt_insecure: bool,
+) -> String {
+    let mut resolved_css = String::from(css_string);
+
+    for link in REGEX_CSS_URL.captures_iter(&css_string) {
+        let target_link = link.name("url").unwrap().as_str();
+
+        // Determine the type of link
+        let is_stylesheet = link.name("stylesheet").is_some();
+        let is_font = link.name("font").is_some();
+        let is_image = !is_stylesheet && !is_font;
+
+        // Generate absolute URL for content
+        let embedded_url = match resolve_url(href, target_link) {
+            Ok(url) => url,
+            Err(_) => continue, // Malformed URL
+        };
+
+        // Download the asset.  If it's more CSS, resolve that too
+        let content = if is_stylesheet {
+            // The link is an @import link
+            retrieve_asset(
+                cache,
+                &embedded_url,
+                false,      // Formating as data URL will be done later
+                "text/css", // Expect CSS
+                opt_user_agent,
+                opt_silent,
+                opt_insecure,
+            )
+            .map(|(content, _)| {
+                resolve_css_imports(
+                    cache,
+                    &content,
+                    true, //NOW, convert to data URL
+                    &embedded_url,
+                    opt_no_images,
+                    opt_user_agent,
+                    opt_silent,
+                    opt_insecure,
+                )
+            })
+        } else if (is_image && !opt_no_images) || is_font {
+            // The link is some other, non-@import link
+            retrieve_asset(
+                cache,
+                &embedded_url,
+                true, // Format as data URL
+                "",   // Unknown MIME type
+                opt_user_agent,
+                opt_silent,
+                opt_insecure,
+            )
+            .map(|(a, _)| a)
+        } else {
+            // If it's a datatype that has been opt_no'd out of, replace with
+            // absolute URL
+
+            Ok(embedded_url.clone())
+        }
+        .unwrap_or_else(|e| {
+            eprintln!("Warning: {}", e,);
+
+            //If failed to resolve, replace with absolute URL
```

Comment:
It seems like rustfmt omits this, but I still would like to have a space after `//` for readability purposes.